docs/visual-basic/programming-guide/language-features/procedures/snippets/how-to-call-an-event-handler/Class1.vb

```diff
@@ -0,0 +1,27 @@
+Public Class SpecialForm : Inherits System.Windows.Forms.Form
+
+    Sub New()
+        OnFormOpen()
+    End Sub
+
+    '<Snippet5>
+    ' Place these procedures inside a Form class definition.
+    Private Sub OnFormClosing(sender As Object, e As System.ComponentModel.CancelEventArgs)
+        ' Insert code to deal with impending closure of this form.
+    End Sub
+
+    Public Sub OnFormOpen()
+        AddHandler Me.FormClosing, AddressOf OnFormClosing
+    End Sub
+    '</Snippet5>
+End Class
```

Comment:
```suggestion
Public Class SpecialForm : Inherits System.Windows.Forms.Form
    '<Snippet5>
    Sub New()
        AddHandler Me.FormClosing, AddressOf OnFormClosing
    End Sub

    Private Sub OnFormClosing(sender As Object, e As System.ComponentModel.CancelEventArgs)
        ' Insert code to deal with impending closure of this form.
    End Sub
    '</Snippet5>
End Class
```

The .ctor should do the work to register the event handler, not call a method. You had a helper method called, `OnFormOpen`. Instead, have the adding of the handler be done from the constructor. Then it can be include in the snippet. The .ctor is now there, and shows the registration of the event and below that the actual event handler.